runtime/databricks/automl_runtime/forecast/utils.py

```diff
@@ -111,3 +112,57 @@ def generate_cutoffs(df: pd.DataFrame, horizon: int, unit: str,
 
 def is_quaterly_alias(freq: str):
     return freq in QUATERLY_OFFSET_ALIAS
+
+def is_frequency_consistency(
+                start_time: Union[pd.Series, pd.Timestamp],
+                end_time: Union[pd.Series, pd.Timestamp], 
+                freq:str) -> bool:
+    """
+    Validate the periods given a start time, end time is consistent with given frequency.
```

Comment:
Could we elaborate what we mean consistent here? I'm afriad we'll forget the definition of consistency when we look back after a few months.
Maybe explicitly saying that we want each timestamp in `end_time` is equal to start_time + k * freq, where k are non negative ints.

Reply:
Done